packages/medusa/src/api/middlewares/check-registered-modules.ts

```diff
@@ -0,0 +1,15 @@
+import { NextFunction, Request, Response } from "express"
+
+export function checkRegisteredModules(services: {
+  [serviceName: string]: string
+}): (req: Request, res: Response, next: NextFunction) => Promise<void> {
+  return async (req: Request, res: Response, next: NextFunction) => {
+    for (const service of Object.keys(services)) {
```

Comment:
todo: should we add a test, I think that in order for the resolve to not throw, we need to add the option `allowUnregistered` otherwise it will not return undefined but throw instead. Can I get you to try please